### PR TITLE
portscan: use network add-on for outgoing proxy

### PR DESCRIPTION
**Pull request by thc202**
Use the network add-on to obtain the outgoing proxy instead of core, the latter is deprecated.

### CODE REVIEW

File: addOns/portscan/portscan.gradle.kts [hunk 1]
```diff
@@ -10,5 +10,20 @@ zapAddOn {
...
+            }
+        }
     }
 }
+
+dependencies {
+    compileOnly(parent!!.childProjects.get("network")!!)
+
+    testImplementation(parent!!.childProjects.get("network")!!)
+    testImplementation(project(":testutils"))
```
Comment (kingthorin) [inline addOns/portscan/portscan.gradle.kts:28]: It doesn't really matter, but any point adding these without some tests?

Comment (thc202) [reply addOns/portscan/portscan.gradle.kts:28]: Copy/paste from other add-on, will remove.

Comment (thc202) [reply addOns/portscan/portscan.gradle.kts]: Done.

**Review by kingthorin — APPROVED** [review]

**Review by ricekot — APPROVED** [review]
Thank you!

**Pull request merged**